refactor: Port all direct CodedString usages to StyledText

Mcrtin opened the pull request:
Ports the remaining usages of CodedString to StyledText.

Mcrtin committed:
refactor: Port ActionBarHandler to StyledText

Mcrtin committed:
refactor: Remove unused CodedString methods in ClientsideMessageEvent

Mcrtin committed:
refactor: Port ContainerSearchFeature to StyledText

Mcrtin committed:
refactor: Remove unused imports

Mcrtin committed:
refactor: Port ItemScreenshotFeature to StyledText

Mcrtin committed:
refactor: Port InventoryFunctions to StyledText

Mcrtin committed:
refactor: Port DiscoveryInfo to StyledText

Mcrtin committed:
refactor: Port WynnItemParser to StyledText

Mcrtin committed:
refactor: Port QuestInfoParser to StyledText

Mcrtin committed:
refactor: Port TowerAuraTimerModel to StyledText

Mcrtin committed:
refactor: Remove unused CodedString methods in ComponentUtils

Mcrtin committed:
fix: ActionBarHandler used StyledText#toSting instead of getSting

Mcrtin committed:
refactor: Port ScoreboardLineChange

Mcrtin committed:
refactor: Port usage of ComponentUtils#getCoded to StyledText#fromComponent

kristofbolyai commented on common/src/main/java/com/wynntils/features/inventory/ContainerSearchFeature.java:
```diff
@@ -146,14 +145,13 @@ private void tryAutoSearch(AbstractContainerScreen<?> abstractContainerScreen) {
 
-        if (!name.getMatcher(currentSearchableContainerType.getNextItemPattern())
-                .matches()) {
+        if (!name.matches(currentSearchableContainerType.getNextItemPattern())) {
```
Did you check all patterns here?

Mcrtin replied:
i did

kristofbolyai commented on common/src/main/java/com/wynntils/features/inventory/ItemScreenshotFeature.java:
```diff
@@ -125,10 +125,10 @@ private void takeScreenshot(Screen screen, Slot hoveredSlot, List<Component> ite
             // First try to save it to disk
-            String itemNameForFile = WynnUtils.normalizeBadString(ComponentUtils.stripFormatting(
-                            CodedString.fromString(itemStack.getHoverName().getString())))
+            String itemNameForFile = WynnUtils.normalizeBadString(StyledText.fromComponent(itemStack.getHoverName())
```
Pretty sure you can do StyledText.fromComponent(..).normalized().getString() instead

Mcrtin replied:
I didn't know they were doing the same thing

kristofbolyai commented on common/src/main/java/com/wynntils/handlers/bossbar/BossBarHandler.java:
```diff
@@ -57,14 +55,18 @@ public void add(
+                    break;
+                }
+            }
+            if (trackedBar == null) return;
```
Why was this changed?

Mcrtin replied:
better readability; should i change it back?

kristofbolyai replied:
Well, I think it is just as readable now as before. And now comes the magic sentence: "no unnecessary changes in porting PRs"

kristofbolyai commented on common/src/main/java/com/wynntils/handlers/scoreboard/type/ScoreboardLineChange.java:
```diff
@@ -4,7 +4,7 @@
 import net.minecraft.server.ServerScoreboard;
 
-public record ScoreboardLineChange(CodedString lineText, ServerScoreboard.Method method, int lineIndex) {}
+public record ScoreboardLineChange(StyledText lineText, ServerScoreboard.Method method, int lineIndex) {}
```
How come this wasn't changed in the scoreboard PR and it still works?

kristofbolyai replied:
Oh so this file is not used. Can you just delete it then?

kristofbolyai commented on common/src/main/java/com/wynntils/models/discoveries/type/DiscoveryType.java:
```diff
@@ -20,10 +20,9 @@ public enum DiscoveryType {
         for (DiscoveryType type : values()) {
-            if (name.getInternalCodedStringRepresentation().charAt(1)
-                    == type.getColor().getChar()) {
+            if (name.getString().charAt(1) == type.getColor().getChar()) {
```
I am usually against changes in a port PR.. However, name.firstPart.getStyle().getColor is probably better to compare here..

Mcrtin replied:
maybe we should add a method for that 
```java
name.getFirstPart().getPartStyle().getStyle().getColor().getValue() == type.getColor().getColor()
```

kristofbolyai commented on common/src/main/java/com/wynntils/models/wynnitem/parsing/WynnItemParser.java:
```diff
@@ -34,19 +33,18 @@
+            "^§[ac]([-+]\\d+)(?:§[24] to §[ac](-?\\d+))?(%| tier|/[35]s)?(?:§8/(\\d+)(?:%| tier|/[35]s)?)?(?:§2(\\*{1,3}))? ?§7 ?(.*)$");
 
     // Test suite: https://regexr.com/782rk
     private static final Pattern TIER_AND_REROLL_PATTERN = Pattern.compile(
-            "^(§fNormal|§eUnique|§dRare|§bLegendary|§cFabled|§5Mythic|§aSet|§3Crafted) ([A-Za-z\\d _]+)(?:§r§8)?(?: \\[(\\d+)(?:\\/(\\d+) Durability)?\\])?$");
+            "^(§fNormal|§eUnique|§dRare|§bLegendary|§cFabled|§5Mythic|§aSet|§3Crafted) ([A-Za-z\\d _]+)(?:§8)?(?: \\[(\\d+)(?:\\/(\\d+) Durability)?\\])?$");
 
     // Test suite: https://regexr.com/778gk
     private static final Pattern POWDER_PATTERN =
-            Pattern.compile("^§7\\[(\\d+)/(\\d+)\\] Powder Slots(?: \\[§r§(.*)§r§7\\])?$");
+            Pattern.compile("^§7\\[(\\d+)/(\\d+)\\] Powder Slots(?: \\[§(.*)§7\\])?$");
 
     // Test suite: https://regexr.com/79atu
-    private static final Pattern EFFECT_LINE_PATTERN =
-            Pattern.compile("^§(.)- §r§7(.*): §r§f([+-]?\\d+)(?:§.§.)? ?(.*)$");
+    private static final Pattern EFFECT_LINE_PATTERN = Pattern.compile("^§(.)- §7(.*): §f([+-]?\\d+)(?:§.§.)? ?(.*)$");
```
Have you tested that items still parse?

kristofbolyai replied:
(also perhaps can you update the tests?)

Mcrtin replied:
i don't know how to update them but the items look the same and there are no errors except
```
[Render thread/WARN] (wynntils) Item literal{§cCharm of the Stone}[style={}] has unknown identified stat Loot from Lv. 90-110 content```

Mcrtin committed:
refactor: Use StyledText#getNormalized instead of WynnUtils#normalizeBadString

Mcrtin committed:
refactor: Use StyledText#getNormalized instead of WynnUtils#normalizeBadString

kristofbolyai commented on common/src/main/java/com/wynntils/utils/mc/ComponentUtils.java:
```diff
@@ -40,22 +41,7 @@ public static CodedString getCoded(Component component) {
-        if (component == null) return null;
-
-        return getUnformatted(component);
+        return StyledText.fromComponent(component).getString(PartStyle.StyleType.NONE);
```
I am not sure if we should do this. Maybe we should just leave this be until the whole method is removed

Mcrtin committed:
Merge remote-tracking branch 'origin/port-to-StyledText' into port-to-StyledText

Mcrtin committed:
Merge branch 'main' into port-to-StyledText

Mcrtin committed:
Undo change in ComponentUtils#getUnformatted;
Delete ScoreboardLineChange;
Port style comparison in DiscoveryType#getDiscoveryTypeFromString

Mcrtin committed:
Merge remote-tracking branch 'origin/port-to-StyledText' into port-to-StyledText

Mcrtin committed:
Undo change in BossBarHandler#add

kristofbolyai reviewed (approved):
Looks good!
